commands.js

```diff
@@ -1741,7 +1741,12 @@ exports.commands = {
 			let roomGroup = (room.auth && room.isPrivate === true ? ' ' : user.group);
 			if (room.auth && user.userid in room.auth) roomGroup = room.auth[user.userid];
 			if (Config.groupsranking.indexOf(target) > Math.max(1, Config.groupsranking.indexOf(roomGroup)) && !user.can('makeroom')) {
-				return this.errorReply("/modchat - Access denied for setting higher than " + Config.groupsranking[1] + ".");
+				let highestModchat = Config.groupsranking[1];
+				if (user.can('declare', null, room)) {
+					let roomAuth = (room.auth && room.auth[user.userid] ? room.auth[user.userid] : false);
```

Comment:
This seems like very different logic.

You should probably merge it with the logic from `Config.groupsranking.indexOf(target) > Math.max(1, Config.groupsranking.indexOf(roomGroup)) && !user.can('makeroom')`